src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -5125,7 +5125,7 @@ def list_generator(pages, num_results=50):
 
         # handle num results
         if num_results is not None:
-            if num_results >= len(result):
+            if num_results <= len(result):
                 break
```

Comment:
The logic here is to stop querying the next batch of data when the number of data queried exceeds the number specified by customer